wolfcrypt/src/asn.c

```diff
@@ -19936,6 +19936,78 @@ int wc_PemPubKeyToDer(const char* fileName,
 
 #endif /* !NO_FILESYSTEM && WOLFSSL_PEM_TO_DER */
 
+/* Get public key in DER format from DER encoded X.509 certificate.
+ *
+ * x509   - input certificate in DER format
+ * x509Sz - size of x509, octets
+ * derKey - output buffer to place DER encoded public key
+ * derKeySz [IN/OUT] - size of derKey buffer on input, size of public key
+ *                     on return. If derKey is passed in as NULL, derKeySz
+ *                     will be set to required buffer size for public key
+ *                     and LENGTH_ONLY_E will be returned from function.
+ * Returns 0 on success, or negative error code on failure. LENGTH_ONLY_E
+ * if derKey is NULL and returning length only.
+ */
+int wc_GetPubKeyDerFromX509Der(const byte* x509, word32 x509Sz,
```

Comment:
Would be handy to support PEM format here also, but a user can always call `wc_CertPemToDer`. Also it would be nice to support the `CERTREQ_TYPE` arg in `ParseCertRelative`.

Reply:
Maybe call it `wc_GetPubKeyDerFromX509`?

Reply:
Great feedback. Added support for `type`, which supports `CERTREQ_TYPE` and `CERT_TYPE`.  Added support for `format` which supports `CTC_FILETYPE_PEM` or `CTC_FILETYPE_ASN1`.  PEM support is protected with `#ifndef WOLFSSL_NO_MALLOC`, as `PemToDer()` uses XMALLOC.  DO-178 will not allow dynamic memory, so this will provide an easy way to cut this functionality out for those users.